src/transformers/models/pop2piano/processing_pop2piano.py

```diff
@@ -0,0 +1,114 @@
+    processor.
+
+    [`Pop2PianoProcessor`] offers all the functionalities of [`Pop2PianoFeatureExtractor`] and [`Pop2PianoTokenizer`].
+    See the docstring of [`~Pop2PianoProcessor.__call__`] and [`~Pop2PianoProcessor.decode`] for more information.
+
+    Args:
+        feature_extractor (`Pop2PianoFeatureExtractor`):
+            An instance of [`Pop2PianoFeatureExtractor`]. The feature extractor is a required input.
+        tokenizer (`Pop2PianoTokenizer`):
+            An instance of ['Pop2PianoTokenizer`]. The tokenizer is a required input.
+    """
+    attributes = ["feature_extractor", "tokenizer"]
+    feature_extractor_class = "Pop2PianoFeatureExtractor"
+    tokenizer_class = "Pop2PianoTokenizer"
+
+    def __init__(self, feature_extractor, tokenizer):
+        super().__init__(feature_extractor, tokenizer)
+
+    def __call__(
+        self,
+        audio: Union[np.ndarray, List[float], List[np.ndarray]] = None,
+        sampling_rate: Union[int, List[int]] = None,
+        steps_per_beat: int = 2,
+        do_infer_resample: Optional[bool] = True,
+        return_attention_mask: Optional[bool] = False,
+        return_tensors: Optional[Union[str, TensorType]] = None,
+        token_ids: Union[List, TensorType] = None,
+        feature_extractor_output: BatchFeature = None,
+        return_midi: bool = True,
+        **kwargs,
+    ) -> BatchFeature:
+        """
+        This method uses [`Pop2PianoFeatureExtractor.__call__`] method to prepare log-mel-spectrograms for the model,
+        and [`Pop2PianoTokenizer.__call__`] to prepare pretty_midi objects from the model outputs.
+
+        Please refer to the docstring of the above two methods for more information.
+        """
+
+        # Since Feature Extractor needs both audio and sampling_rate and tokenizer needs both token_ids and
+        # feature_extractor_output, we must check for both.
+        if (audio is None and sampling_rate is None) and (token_ids is None and feature_extractor_output is None):
+            raise ValueError(
+                "You have to specify at least audios and sampling_rate in order to use feature extractor or "
+                "token_ids along with feature_extractor_output to use the tokenizer part."
+            )
+
+        encoding = BatchFeature()
+
+        if audio is not None and sampling_rate is not None:
+            extracted_features = self.feature_extractor(
+                audio=audio,
+                sampling_rate=sampling_rate,
+                steps_per_beat=steps_per_beat,
+                do_infer_resample=do_infer_resample,
+                return_attention_mask=return_attention_mask,
+                return_tensors=return_tensors,
+            )
+            encoding.update(extracted_features)
+
+        if token_ids is not None and feature_extractor_output is not None:
+            tokenizer_outputs = self.tokenizer(
```

Comment:
In this case, the tokenizer should be mapping from tokens to ids when the processor is called e.g. like for [wav2vec2](https://github.com/huggingface/transformers/blob/main/src/transformers/models/wav2vec2/processing_wav2vec2.py)